ui/qt-lib/src/Ariadne/UI/Qt/Widgets/Dialogs/ConfirmMnemonic.hs

```diff
@@ -0,0 +1,233 @@
+module Ariadne.UI.Qt.Widgets.Dialogs.ConfirmMnemonic
+  ( runConfirmMnemonic
+  ) where
+
+import Universum
+
+import qualified Data.Text as T
+import Formatting
+
+import Graphics.UI.Qtah.Core.Types
+  (QtCursorShape(IBeamCursor), QtTextInteractionFlag(..))
+import Graphics.UI.Qtah.Signal (connect_)
+
+import qualified Graphics.UI.Qtah.Gui.QCursor as QCursor
+import qualified Graphics.UI.Qtah.Widgets.QAbstractButton as QAbstractButton
+import qualified Graphics.UI.Qtah.Widgets.QBoxLayout as QBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QCheckBox as QCheckBox
+import qualified Graphics.UI.Qtah.Widgets.QDialog as QDialog
+import qualified Graphics.UI.Qtah.Widgets.QHBoxLayout as QHBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QLabel as QLabel
+import qualified Graphics.UI.Qtah.Widgets.QLineEdit as QLineEdit
+import qualified Graphics.UI.Qtah.Widgets.QPushButton as QPushButton
+import qualified Graphics.UI.Qtah.Widgets.QWidget as QWidget
+
+import Ariadne.UI.Qt.UI
+import Ariadne.UI.Qt.Widgets.Dialogs.Util
+
+data ConfirmationState
+  = Before
+  | DisplayMnemonic
+  | RetypeMnemonic
+  | Done
+  deriving (Eq)
+
+data ConfirmMnemonic =
+  ConfirmMnemonic
+    { confirmMnemonic :: QDialog.QDialog
+    , mnemonic :: [Text]
+    , instructionLabel :: QLabel.QLabel
+    , noOneLooks :: QCheckBox.QCheckBox
+    , retypeMnemonic :: QLineEdit.QLineEdit
+    , clearButton :: QPushButton.QPushButton
+    , actionButton :: QPushButton.QPushButton
+    , confirmationState :: IORef ConfirmationState
+    , noOneLooksWidget :: QWidget.QWidget
+    , mnemonicWidget :: QLabel.QLabel
+    , retypeWidget :: QWidget.QWidget
+    , checksWidget :: QWidget.QWidget
+    , uMoneyOnDevice :: QCheckBox.QCheckBox
+    , uMoved :: QCheckBox.QCheckBox
+    }
+
+initConfirmMnemonic :: [Text] -> IO ConfirmMnemonic
+initConfirmMnemonic mnemonic = do
+  confirmMnemonic <- QDialog.new
+  layout <- createLayout confirmMnemonic
+
+  let headerString :: String = "RECOVERY PHRASE"
+
+  QWidget.setWindowTitle confirmMnemonic headerString
+
+  header <- QLabel.newWithText headerString
+  addHeader layout header
+
+  instructionLabel <- QLabel.new
+  QLabel.setWordWrap instructionLabel True
+  QWidget.setMinimumSizeRaw instructionLabel 600 30
+  QBoxLayout.addWidget layout instructionLabel
+
+  (noOneLooksWidget, noOneLooksLayout) <- createSubWidget
+  noOneLooks <- createCheckBox noOneLooksLayout
+    "Make sure nobody looks into your screen unless you want them to have access to your funds."
+  QBoxLayout.addWidget layout noOneLooksWidget
+
+  mnemonicWidget <- QLabel.new
+  void $ setProperty mnemonicWidget ("styleRole" :: Text) ("mnemonicDisplay" :: Text)
+  QLabel.setTextInteractionFlags mnemonicWidget TextSelectableByMouse
```

Comment:
I believe allowing selecting (and therefore copying) is a good think to do, right?